return {
	["_locale"] = "en_US",
	["_locale_display"] = "English (United States)",
	["_locale_display_localized"] = "English (United States)",

	["Apps.MainFrame.Config.Desc"] = "Configure Administer, admins, and install new apps.",
	["Apps.MainFrame.Config.Title"] = "Configuration",
	["Apps.MainFrame.Home.Desc"] = "Everything you need to see at a glance.",
	["Apps.MainFrame.Home.Title"] = "Home",
	["Apps.MainFrame.Template.Desc"] = "Your startpage with everything you need.",
	["Configuration.Admins.Container.Admins.Content.Template.Info"] = "Administrator (Rank 1)",
	["Configuration.Admins.Container.Admins.Content.Template.Metadata"] = "Added by @username <b>0 weeks ago</b>",
	["Configuration.Admins.Container.Admins.Header.TLabel"] = "Admins",
	["Configuration.Admins.Container.Ranks.Content.Template.Info"] = "Rank %n • %n pages • {{ Protected | Not Protected }}",
	["Configuration.Admins.Container.Ranks.Header.TLabel"] = "Ranks",
	["Configuration.Admins.EditRank.Apps.Content.Template.AppName"] = "Configuration",
	["Configuration.Admins.EditRank.Apps.Header.TLabel"] = "Apps",
	["Configuration.Admins.EditRank.Header.Finish.HeaderLabel"] = "FINISH",
	["Configuration.Admins.EditRank.Header.TLabel"] = "Editing <rank>",
	["Configuration.Admins.EditRank.People.Content.GroupTemplate.Label"] = "Group",
	["Configuration.Admins.EditRank.People.Content.GroupTemplate._Name"] = "Group (0 members)",
	["Configuration.Admins.EditRank.People.Content.PlayerTemplate.Label"] = "User",
	["Configuration.Admins.EditRank.People.Content.PlayerTemplate._Name"] = "User (@user)",
	["Configuration.Admins.EditRank.People.Header.TLabel"] = "Members",
	["Configuration.Admins.EditRank.RankID"] = "Rank 0 · Created by <user>",
	["Configuration.Admins.NewAdmin.BottomData.RankTitle"] = "Creating a rank",
	["Configuration.Admins.NewAdmin.BottomData.StepLabel"] = "Step 1/5",
	["Configuration.Admins.NewAdmin.Page1.Body"] = "The Rank Creator allows you to create new admin ranks and specify their permitted apps and users. New users you add will be given access to Administer instantly even if already in-game.You'll only be able to add apps which you personally have access to. Only people with the permission will be able to enabled the option to permit all apps.",
	["Configuration.Admins.NewAdmin.Page1.Header"] = "Welcome to the Rank Creator",
	["Configuration.Admins.NewAdmin.Page1.NextPage.LogoWordmark"] = "Continue",
	["Configuration.Admins.NewAdmin.Page2.FilterHdr"] = "Choose the rank's internal name. Some apps may display this publicly, so it may be filtered.",
	["Configuration.Admins.NewAdmin.Page2.LogoWordmark"] = "Name",
	["Configuration.Admins.NewAdmin.Page2.PreviewText"] = "Preview: ",
	["Configuration.Admins.NewAdmin.Page3.AddPane.AddGroup.LogoWordmark"] = "Group",
	["Configuration.Admins.NewAdmin.Page3.AddPane.AddUser.LogoWordmark"] = "User",
	["Configuration.Admins.NewAdmin.Page3.LogoWordmark"] = "Members",
	["Configuration.Admins.NewAdmin.Page3.PageHdr"] = "Choose the members for this rank. You can edit this later.",
	["Configuration.Admins.NewAdmin.Page4.Apps.Apps.Template.AppDesc"] = "This is a template you shouldn't see.",
	["Configuration.Admins.NewAdmin.Page4.Apps.Apps.Template.AppName"] = "My App",
	["Configuration.Admins.NewAdmin.Page4.Apps.Apps.Template.Status"] = "Showing",
	["Configuration.Admins.NewAdmin.Page4.PageHdr"] = "Choose the apps this rank will be able to use. You may not be able to add any apps that you cannot use.",
	["Configuration.Admins.NewAdmin.Page5.Body"] = "Great! Your rank has been successfully created. Anybody you just added will be given the panel instantly. No need to do anything from here. If you would like to edit this rank in the future, come back to this page at any time and click the settings icon next to the name of the rank you just created.",
	["Configuration.Admins.NewAdmin.Page5.Header"] = "Success!",
	["Configuration.Apps.Content.Template.AppName"] = "Example App",
	["Configuration.Apps.Content.Template.AppShortDesc"] = "Short description, likely for credits/copyright",
	["Configuration.Apps.Content.Template.Delete.HeaderLabel"] = "DELETE",
	["Configuration.Apps.Content.Template.InstallDate"] = "Installed 4 weeks ago",
	["Configuration.Apps.Content.Template.Local.HeaderLabel"] = "Local apps cannot be modified.",
	["Configuration.Apps.Content.Template.Settings.HeaderLabel"] = "OPTIONS",
	["Configuration.Apps.Header.Desc"] = "See the apps you have installed here, and enable/disable them. If they have settings, you can edit them here too.",
	["Configuration.Apps.MessageBox.Content"] = "Would you like to uninstall this app? You will have to reinstall it from it's App Server or from its ID.",
	["Configuration.Apps.MessageBox.Header"] = "Remove {{ app }}?",
	["Configuration.Apps.Options.Frame.HeaderLabel"] = "Configure App",
	["Configuration.Apps.Options.Page.BoolTemplate.Action"] = "Enable",
	["Configuration.Apps.Options.Page.Empty.Description"] = "This app doesn't appear to have any configurable settings for you to play with.",
	["Configuration.Apps.Options.Page.InputTemplate.SettingName"] = "Setting",
	["Configuration.ErrorLog"] = "Error",
	["Configuration.ErrorLog.ScrollingFrame.Template"] = "Warning",
	["Configuration.InfoPage.Credits.log"] = "Administer is powered by many open-source resources.",
	["Configuration.InfoPage.Credits.title"] = "Credits",
	["Configuration.InfoPage.Donate.Buttons.Temp.Label"] = "DONATE %s",
	["Configuration.InfoPage.Donate.Message"] = "Administer is on a mission to develop the best resource and additions to it for free. We've been going for a long time and our work continues to be free. Please donate if Administer adds value to your game or staff team! <b>Your game gets 10% back! Donations are divided among Administer staff.</b>",
	["Configuration.InfoPage.Donate.title"] = "Donations",
	["Configuration.InfoPage.VersionDetails.Credit"] = "Administer Team 2022 - 2024",
	["Configuration.InfoPage.VersionDetails.Update.Label"] = "CHECK FOR UPDATES",
	["Configuration.InfoPage.VersionDetails.Version"] = "Version x (1/2/34)",
	["Configuration.Marketplace.Header.Head"] = "Marketplace",
	["Configuration.Marketplace.Header.Plugins"] = "Apps",
	["Configuration.Marketplace.Header.Themes"] = "Themes",
	["Configuration.Marketplace.Initial.Failure.Header"] = "Key not generated",
	["Configuration.Marketplace.Initial.Failure.Why"] = 'This app server rejected the request to create a key for: "Account distributed keys related to one or more abuse case". To get an old key, please reach out.',
	["Configuration.Marketplace.Initial.HeaderLabel"] = "Hi, <b>user!</b> Just one more step before you get going.",
	["Configuration.Marketplace.Initial.Register.HeaderLabel"] = "REGISTER",
	["Configuration.Marketplace.Install.Head.HeaderLabel"] = "<b>Install App</b> - Moderation+",
	["Configuration.Marketplace.Install.MetaCreated.Label"] = "Created on <b>{{ }}</b>",
	["Configuration.Marketplace.Install.MetaInstalls.Label"] = "<b>{{ }}</b> install",
	["Configuration.Marketplace.Install.MetaServer.Label"] = "Shown because <b>You're subscribed to administer.notpyx.me</b>",
	["Configuration.Marketplace.Install.MetaUpdated.Label"] = "Last Updated <b>1 hour ago</b>",
	["Configuration.Marketplace.Install.MetaVersion.Label"] = '<font color="rgb(139,139,139)" transparency="0">Your version </font><font color="rgb(56,218,111)" transparency="0">is supported! ({{ }})</font>',
	["Configuration.Marketplace.Install.TagsLabel"] = "TAGS",
	["Configuration.Marketplace.Install.UnverifiedWarning"] = "<b>Warning</b>: This app comes from an unverified server. If you install this app you are giving <b>unrestricted serverside code execution</b> to the developer of the app, which can contain viruses or steal your game, make sure you really trust the owner of this server!",
	["Configuration.Marketplace.MPFrozen.HeaderLabel"] = "Oops!",
	["Configuration.Marketplace.MPFrozen.Subheading1"] = "Sorry, but one or more app servers returned an error while processing that (code: {{ }}, route {{ }}). This may be a ban, a temporary ratelimit, or it may be unavailbable. Please retry your request again soon. If you keep seeing this page please check the log and remove any defective app servers.",
	["Configuration.Marketplace.MainMarketplace.Content.Template.AppName"] = "My App",
	["Configuration.Marketplace.MainMarketplace.Content.Template.InstallLabel"] = "MORE",
	["Configuration.Marketplace.MainMarketplace.Content.Template.ShortDesc"] = "Missing description here for some reason...",
	["Configuration.Marketplace.Searchbar.Searchbar.Label"] = "Type an App ID, name, or app server URL...",
	["Configuration.Marketplace.Searchbar.Searchbar.Popup.Label"] = "Type a command...",
	["Configuration.MenuBar.New.AInfoPage.HeaderLabel"] = "Information",
	["Configuration.Settings.CategoriesPage.BoolTemplate.Action.Label"] = "DISABLE",
	["Configuration.Settings.Header.Head"] = "Configure Administer for everybody here.",
	["Configuration.Settings.Options.DetailsCard.AppName"] = "Manage {{ }}",
	["Configuration.Settings.Options.DetailsCard.AppShortDesc"] = "4 settings. Last saved on 10/21/24 at 10:00 PM by @pyxfluff",
	["Header.AppDrawer.HeaderLabel"] = "APPS",
	["Header.Mark.HeaderLabel"] = "<b>Administer</b> · Home",
	["Header.Minimize.HeaderLabel"] = "EXIT",
	["Home.Welcome"] = '<stroke color="rgb(0,0,0)" transparency = "0.85" thickness=".4">Administer is loading...</stroke>',
	["Home.Widget1.Editing.Last"] = "Last",
	["Home.Widget1.Editing.Next"] = "Next",
	["Import.Select.Header.Head"] = "Let's help ease the transition. Select what to import from where.",
	["NotFound.ErrorContent.TextLabel"] = "If you're looking for a menu, it's not here.",
	["NotFound.ErrorContent.sending"] = "This app ({APP_NAME}) didn't link something up correctly. This used to be caused by incorrect code, but as of recently it shouldn't be possible. If you are seeing this, report it to the developer.",
	["NotFound.PageBroken.TextLabel"] = "This page did something that Roblox or Administer could not understand. It cannot continue loading.",
	["NotFound.PageBroken.sending"] = "This app ({APP_NAME}) had some form of runtime or Administer API error, which is fatal and would freeze something otherwise. Please report this error to the developer.",
	["Notepad.Header.Head"] = "Notepad",
	["Welcome.Initial.HeaderLabel"] = "Good morning, <b>Roblox</b>! Welcome to Administer.",
	["Welcome.Initial.Subheading1"] = "You're going to like it here. Welcome to the admin panel with everything you need and none of what you don't.",
	["administer..widgets.NotFound.Welcome"] = "It probably got uninstalled. Please select another one.",
	["administer..widgets.NotSelected.Welcome"] = "Select an app to display here.",
	["administer..widgets.WelcomeMsg.Welcome"] = "To get started with widgets, just install some apps and then click the edit button.",

	["admincheck.OVERRIDE"] = "Found in AdminID Override",
	["admincheck.SUDO"] = "Sandbox mode (sudo) is enabled in your game settings or Variables script.",
	["admincheck.RANKINDEX"] = "Found in the ranks index",
	["admincheck.GROUP"] = "Data based on group rank",
	["admincheck.nil"] = "User not found in the admin index",
	["admin.RankDefault"] = "Administrator",
	["admin.NoRank"] = "Non Admin",
	["admin.InOverride"] = "This user is in the override module, as such we don't have any information.",
	["admin.group.AllRanks"] = "All Ranks",
	["admin.rv2.Migrate"] = "Sorry, but you may not create new ranks before updating to Ranks V2.",
	["admin.rankcreate.MessagingServiceFailed"] = "We made the rank fine, but failed to publish the event to tell other servers to check. Please try freeing up some MessagingService slots. {e}",
	["admin.rankcreate.Done"] = "Successfully made a rank!",
	["admin.ranktype.Protected"] = "Protected ranks cannot be edited by anybody but the system.",
	["admin.ranktype.Protected.meta"] = "This rank is protected.",
	["admin.rankcreator.BadData"] = "You must provide a valid AdminRank!",
	["admin.rank.Default"] = "Added by System for first-time setup.",
	["admin.rankcreator.error.Generic"] = "`We couldn't process that request right now, try again later.\n\n%s`",
	["admin.rank.created"] = "Created this from the rank editor.",

	["generic.Unauthorized"] = "Unauthorized",
	["generic.error.CheckLogs"] = "Something went wrong, please refer to the logs.",
	["generic.error.Generic"] = "Something went wrong.",
	["generic.error.TryAgainLater"] = "Try again later.",
	["generic.Please"] = "Please",
	["generic.ThankYou"] = "Thank you!",
	["generic.Created"] = "Created",
	["generic.Done"] = "Done",
	["generic.Updated"] = "Updated",
	["generic.Install"] = "Install",
	["generic.Load"] = "Load",
	["generic.Loading"] = "Loading",
	["generic.Group"] = "Group",
	["generic.User"] = "User",
	["generic.Yes"] = "Yes",
	["generic.No"] = "No",
	["generic.OK"] = "OK",
	["generic.UI"] = "UI",
	["generic.Cancel"] = "Cancel",
	["generic.Confirm"] = "Confirm",
	["generic.Download"] = "Download",
	["generic.Remove"] = "Remove",
	["generic.Protected"] = "Protected",
	["generic.admin"] = "Admin",
	["generic.Forward"] = "Next",
	["generic.Exit"] = "Exit",

	["onboarding.Welcome"] = "%s loaded! You're a(n) %s. Press %s to enter the panel.",
	["onboarding.Head"] = "Welcome to %s!",

	["apps.NoFetch"] = "Failed to fetch apps!",
	["apps.BadServer"] = "This is not a valid Administer app server.",
	["apps.Disable"] = 'You can re-enable it from the "Disabled Apps" menu. The app may be able to continue running for this session but it will not be started in any new servers.',
	["apps.Remove"] = "This app will not start in any new servers but will continue running.",
	["apps.DupeDetected"] = "You may only install an app once!",
	["apps.Downloaded"] = "Successfully installed an app!",
	["apps.BadID"] = "Bad app ID! This app server is returning bad data.",
	["apps.server.Down"] = "Failed reaching out to the App Server. Perhaps it's offline?",
	["apps.server.NotAdminister"] = "Invalid app server! Check logs for more info.",
	["apps.pulled.Bad"] = "Something went wrong with this module: %s",

	["betas.Mobile.body"] = "You've successfully opted in to the Administer Mobile Beta.",
	["betas.Mobile.ctx"] = "Mobile Beta",

	["error.NoPing"] = "Administer server ping failed, it seems your client may be incorrectly installed or the server si not executing properly. Please reinstall from source.",
	["error.StartupFailed"] = "Startup Failed",
	["error.LostDep"] = "Failed to boot client, missing required dependency (%s), please reinstall or roll back any recent changes",
	["error.LostDep.ClientFacingError"] = "Startup aborted, please make sure Administer is correctly installed. (failed dependency: %s)",
	["error.BootFail"] = "Boot failure",

	["update.Check"] = "Check for Updates",
	["updater.Failed.Body"] = "This might be an issue with online services, please try again later; code %s",
	["updater.Failed.Header"] = "Version check failed",

	["donations.Done"] = "Thank you for your support, %s! Your donation helps ensure future Administer updates for years to come ^^",
	["donations.DonationProcessed"] = "Your donation helps give back to the open-source community and will ensure: - lots of updates in the future - development funding for more apps - app server hosting Your donation means a lot to us! Make sure to show us this to claim your donor perks.",
	["donations.BadProxy"] = "The proxy may be offline or experiencing issues.",

	["updater.outofdate.Body"] = "%s is out of date. Please update your module.",
	["updater.outofdate.Header"] = "Version check complete",

	["filter.failed"] = "Failed to filter!",

	["setting.Admins.CheckTime.DisplayName"] = "Admin check time",
	["setting.Admins.CheckTime.Description"] = "How often to check if admins are still admins, and take away the panel if they are not.",

	["setting.Admins.AllowProtectedEdits.DisplayName"] = "Edit Protected Ranks",
	["setting.Admins.AllowProtectedEdits.Description"] = "Allows editing protected ranks (superadmin). Also allows the creation of them. Not recommended as you could potentially break your installation.",

	["setting.Interface.AnimationSpeed.DisplayName"] = "Animation Speed",
	["setting.Interface.AnimationSpeed.Description"] = "Controls the speed of client side animations. Higher is slower. Set to 0 to disable animations.",

	["setting.Interface.DisableUnofficialAppWarning.DisplayName"] = "Disable unofficial app warning",
	["setting.Interface.DisableUnofficialAppWarning.Description"] = "Hides the warning which says if a plugin is external. Useful for new people, otherwise just UI clutter.",

	["setting.Interface.EnableEditableImages.DisplayName"] = "Enable EditableImages",
	["setting.Interface.EnableEditableImages.Description"] = 'Enables EditableImage features such as app card backgrounds, app card reflectios, and other blurring effects. Does not disable "image glow". Currently nonfunctional outside of Studio due to Roblox limitations.',

	["setting.Interface.HomepageGreeting.DisplayName"] = "Homepage Greeting",
	["setting.Interface.HomepageGreeting.Description"] = 'This is the text that displays after the "Good morning, username!" text on the homepage.',
	["setting.Interface.HomepageGreeting.Value"] = "Welcome to Administer!",

	["setting.Interface.PanelKeybind.DisplayName"] = "Panel Keybind",
	["setting.Interface.PanelKeybind.Description"] = "The key used to open the panel on keyboard-enabled devices.",

	["setting.Interface.EnableClickEffects.DisplayName"] = "Enable button effects",
	["setting.Interface.EnableClickEffects.Description"] = "Increases memory usage slightly, also makes noise.",

	["setting.Apps.LoadDelay.DisplayName"] = "LoadDelay",
	["setting.Apps.LoadDelay.Description"] = "Enables a 3s delay before loading any admins. Use this solve apps not initializing.",
	["setting.Apps.LoadDelay.Value"] = "None",

	["setting.Apps.FetchAppServers.DisplayName"] = "FetchAppServers",
	["setting.Apps.FetchAppServers.Description"] = "Disallows Administer from contacting any external app servers.",

	["setting.Apps.LoadApps.DisplayName"] = "LoadApps",
	["setting.Apps.LoadApps.Description"] = "Controls the ability for the server to load external apps. You shouldn't enable this unless you need a clean install or need to remove it.",

	["setting.Accessibility.ChatCommand.DisplayName"] = "ChatCommand",
	["setting.Accessibility.ChatCommand.Description"] = "Enables an /adm command to open Administer.",

	["setting.Accessibility.DisplayHours.DisplayName"] = "DisplayHours",
	["setting.Accessibility.DisplayHours.Description"] = "If enabled, hours will be displayed on all timestamps, may be ignored in some.",

	["setting.Accessibility.ShortNumberDecimals.DisplayName"] = "ShortNumberDecimals",
	["setting.Accessibility.ShortNumberDecimals.Description"] = "How many decimals to display next to short numbers. 2 -> 12.34k, 3 -> 12.345k, etc.",

	["setting.Accessibility.TopbarPlus.DisplayName"] = "TopbarPlus",
	["setting.Accessibility.TopbarPlus.Description"] = "Enables a Topbar+ section that you can launch Administer and apps from.",

	["setting.Accessibility.MobileOpenGesture.DisplayName"] = "MobileOpenGesture",
	["setting.Accessibility.MobileOpenGesture.Description"] = "Controls how big the mobile open gesture is from the right side. If the value is disabled you will need either the chat command or Topbar+ setting enabled.",
	["setting.Accessibility.MobileOpenGesture.Value"] = "regular",

	["setting.Miscellaneous.EditableImageRenderingDelay.DisplayName"] = "EditableImageRenderingDelay",
	["setting.Miscellaneous.EditableImageRenderingDelay.Description"] = "Enables a task.wait() call in EditableImage rendering calls. Can help reduce significant lag if present. WARNING: Makes client loading SIGNIFICANTLY slower.",

	["setting.Miscellaneous.NotificationCloseTimer.DisplayName"] = "NotificationCloseTimer",
	["setting.Miscellaneous.NotificationCloseTimer.Description"] = "Default waiting time to close the notification at. Can be overridden by notifications.",

	["setting.Miscellaneous.SandboxMode.DisplayName"] = "sudo",
	["setting.Miscellaneous.SandboxMode.Description"] = "(studio-only) Gives all people who join the game rank 1 (superadmin). Useful for setting up. Once your ranks are set up, you should turn this off. (also known as sudo)",

	["setting.Miscellaneous.Verbose.DisplayName"] = "Verbose",
	["setting.Miscellaneous.Verbose.Description"] = "Prints out general log data, useful for technical users or debugging. If enabled, it'll appear in the Log too.",

	["setting.category.Admins"] = "Admins",
	["setting.category.Interface"] = "Interface",
	["setting.category.Apps"] = "Apps",
	["setting.category.Accessibility"] = "Accessibility",
	["setting.category.Miscellaneous"] = "Miscellaneous",
}
